engines/qdengine/qdcore/qd_game_object_moving.cpp

```diff
@@ -638,92 +645,94 @@ Vect3f qdGameObjectMoving::get_future_r(float dt, bool &end_movement, bool real_
 
 	qdGameObjectState *st;
 
-	switch (_movement_mode) {
-	case MOVEMENT_MODE_START:
-		if (real_moving) {
-			_movement_mode_time_current += dt;
-			if (_movement_mode_time_current >= _movement_mode_time) {
-				_movement_mode = MOVEMENT_MODE_MOVE;
-				set_walk_animation();
-				get_animation()->set_time_rel(0.f);
+	if (g_engine->_gameVersion > 20030919) {
```

Comment:
I would do it differently.

1. Add `MOVEMENT_MODE_NONE_EARLY`, assign it to `_movement_mode` on loading/creating these object
2. Here add `case MOVEMENT_MODE_NONE_EARLY: // _gameVersion <= 20030919` next to `default:` label, e.g. no-op
3. at the end `if (_movement_mode != MOVEMENT_MODE_NONE_EARLY) _movement_mode = MOVEMENT_MODE_MOVE`